metadata/metadata.go

```diff
@@ -91,15 +92,19 @@ func Pairs(kv ...string) MD {
 }
 
 // String implements the Stringer interface for pretty-printing a MD.
-// Ordering of the values is non-deterministic as it ranges over a map.
 func (md MD) String() string {
 	var sb strings.Builder
 	fmt.Fprintf(&sb, "MD{")
-	for k, v := range md {
+	keys := make([]string, 0, len(md))
+	for k := range md {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+	for _, k := range keys {
```

Comment:
Use a `first` bool instead of relying on `sb.Len() > 3`.

Reply:
Done!